### PR TITLE
Add RMST test

### LINKED ISSUE #16: add RMST
add RMST into `simtrial`

### PR DESCRIPTION
**Pull request by LittleBeannie**
Closes #16

### CODE REVIEW

**Comment by LittleBeannie** [comment]
The current `rmst()` function asks users to input the variable name of the tte, event indicator, grouping variable, such as 
```
lung |> 
  rmst(var_label_tte = "time",
       var_label_event = "status",
       var_label_group = "sex",
       tau = 10,
       reference = "female")
```

By mimic the `survival` R package, a better syntax may be 
```
rmst(Surv(time, status) ~ sex, data = lung)
```
Any suggestions/comments?

References: `survdiff(Surv(time, status) ~ sex, data = lung)`

**Review by nanxstats — CHANGES_REQUESTED** [review]
Nice. I rearranged the order of functions in `rmst.R` to put the exported functions first and then followed the call sequence, as recommended by the [tidyverse style guide](https://style.tidyverse.org/package-files.html#organisation-1):

> In a file that contains multiple functions, public functions and their documentation should appear first, with private functions appearing after all documented functions.

---

Can you please fix the following (small) issues before we merge? Thanks.

1. `df` is defined in `one_rmst()` but not actually used. Can this be removed or is it intended to be used?
1. The description of "A list of 3 data frames of RMST calculations (`RMST`, `RMSTDIFF`, `ALL`)" does not match the implementation.
1. What does the "std" mean in "Variance, std, and CIs of the estimated RMST"? Maybe rephrase to make it clear.

**Comment by LittleBeannie** [comment]
> Nice. I rearranged the order of functions in `rmst.R` to put the exported functions first and then followed the call sequence, as recommended by the [tidyverse style guide](https://style.tidyverse.org/package-files.html#organisation-1):
> 
> > In a file that contains multiple functions, public functions and their documentation should appear first, with private functions appearing after all documented functions.
> 
> Can you please fix the following (small) issues before we merge? Thanks.
> 
> 1. `df` is defined in `one_rmst()` but not actually used. Can this be removed or is it intended to be used?
> 2. The description of "A list of 3 data frames of RMST calculations (`RMST`, `RMSTDIFF`, `ALL`)" does not match the implementation.
> 3. What does the "std" mean in "Variance, std, and CIs of the estimated RMST"? Maybe rephrase to make it clear.

Hello @nanxstats , thank you for these valuable comments! I have addressed them in my latest commits. Please take a moment to review them and let me know if they are clear to you. Thank you!

**Review by nanxstats — APPROVED** [review]
@LittleBeannie Looking great! Let's merge this 😎
